src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(T element) {
+            this.element = element;
+        }
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. Also, don't use getters and setters in class Node as we have access to private fields of the inner class, so using getters or setters is redundant.